index.js

```diff
@@ -82,13 +83,24 @@ module.exports = {
       this._policyString = buildPolicyString(config.policy);
     }
 
-    // provide configuration needed at run-time for FastBoot support (if needed)
-    // TODO: only inject if application uses FastBoot
-    //       https://github.com/rwjblue/ember-cli-content-security-policy/issues/116
-    if (!this._config.enabled || !this._config.delivery.includes('header')) {
+    // CSP header should only be set in FastBoot if
+    // - addon is enabled and
+    // - configured to deliver CSP via header and
+    // - application has ember-cli-fastboot dependency.
+    this._needsFastBootSupport = this._config.enabled &&
+      this._config.delivery.includes('header') &&
+      // TODO: check if application has ember-cli-fastboot-dependency
+      // https://github.com/rwjblue/ember-cli-content-security-policy/issues/116
+      true;
```

Comment:
Using `ember-cli-version-checker` to check if `ember-cli-fastboot` exists should be straight forward here. Actually I was working on that one before facing #117. But I like to do it in another PR to not overcomplex this one. Also I fear that I will face testing issues caused by `ember-cli-version-checker` and `ember-cli-addon-tests` not playing well together...